### PR TITLE
fix: skia textbox overlay not disappearing on unfocus when FeatureConfiguration changes

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): closes #

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Review by MartinZikmund — APPROVED** [review]
Please also add some simple test - e.g. start with overlay text box, load it, change feature flag, unload the original text box and verify the overlay layer has 0 children

File: src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_TextBox.skia.cs [hunk 1]
```diff
@@ -3148,6 +3149,34 @@ public async Task When_Ctrl_Delete_Undo_Redo()
...
+			};
+
+			await UITestHelper.Load(SUT);
+
+			SUT.Focus(FocusState.Programmatic);
+			await WindowHelper.WaitForIdle();
+
+			Assert.AreEqual(0, SUT.TextBoxView.DisplayBlock.Opacity);
+
+			FeatureConfiguration.TextBox.UseOverlayOnSkia = false;
```
Comment (jeromelaban) [inline src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_TextBox.skia.cs:3173]: Would you need to have a reversed test when the overlay box is enabled by default in your other PR?

Comment (ramezgerges) [reply src/Uno.UI.RuntimeTests/Tests/Windows_UI_Xaml_Controls/Given_TextBox.skia.cs:3173]: I assume you mean disabled* by default? The test sets `UseOverlayOnSkia` to true and restores it afterwards, so it the default value shouldn't matter anyway.

**Comment by jeromelaban** [comment]
@mergifyio backport release/stable/5.1

**Pull request merged**